flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/data/binary/BinaryStringDataUtil.java

```diff
@@ -964,6 +964,14 @@ private static BinaryStringData trimLeftSlow(BinaryStringData str) {
         }
     }
 
+    public static boolean isEmpty(BinaryStringData str) {
+        if (str.javaObject != null) {
+            return str.javaObject.isEmpty();
+        } else {
+            return str.binarySection == null || str.binarySection.getSizeInBytes() == 0;
```

Comment:
We'd better add a comment here to explain why call `str.binarySection.getSizeInBytes()` directly instead of `BinaryStringData#getSizeInBytes`, and also add a test for this utility method.

Reply:
Test will be in `BinaryStringDataTest#testIsEmpty`